cookbook/case_studies/ml_training/nlp_processing/word2vec_and_lda.py

```diff
@@ -0,0 +1,331 @@
+    model = Word2Vec.load(model_ser.path)
+    wv = model.wv
+    print(f"Word vector for {word}:{wv[word]}")
+    print(f"Most similar words in corpus to {word}: {wv.most_similar(word, topn=10)}")
+
+
+# %%
+# Sentence Similarity
+# ========
+#
+# This task computes the Word Mover’s Distance (WMD) metric using the trained embeddings of words. This
+# enables us to assess the distance between two documents in a meaningful way even when they have
+# no words in common. WMD is larger for two completely unrelated sentences and smaller for two closely related
+# sentences. We have already chosen two similar sentences for comparison so the word movers distance
+# should be a low value. You can try altering either ``SENTENCE_A`` or ``SENTENCE_B`` variables to be dissimilar
+# to the other sentence, and check that the value computed is larger.
+@task(cache_version="1.0", cache=True, limits=Resources(mem="200Mi"))
+def word_movers_distance(model_ser: FlyteFile[MODELSER_NLP]) -> float:
+    sentences = [SENTENCE_A, SENTENCE_B]
+    results = []
+    for i in sentences:
+        result = [w for w in utils.tokenize(i) if w not in STOPWORDS]
+        results.append(result)
+    model = Word2Vec.load(model_ser.path)
+    distance = model.wv.wmdistance(*results)
+    print(f"Computing word movers distance for: {SENTENCE_A} and {SENTENCE_B} ")
+    print(f"Word Movers Distance is {distance}")
+    return distance
+
+
+# %%
+# Dimensionality Reduction and Plotting
+# ========
+#
+# This helper function creates a word embeddings matplotlib plot
+# rendered using FlyteDeck and output as html.
+def plot_with_matplotlib(x, y, labels):
+    fig = plt.figure()
+    ax = fig.add_subplot(1,1,1)
+    ax.scatter(x, y)
+    indices = list(range(len(labels)))
+    selected_indices = random.sample(indices, 25)
+    for i in selected_indices:
+        plt.annotate(labels[i], (x[i], y[i]))
+    flytekit.Deck("Word Embeddings", mpld3.fig_to_html(fig))
+
+
+# %%
+# The word embeddings made by the model can be visualised by reducing
+# dimensionality of the words to 2 dimensions using tSNE.
+# This task calls the ``plot_with_matplotlib`` function to generate the word
+# embeddings plot after dimensionality reduction.
+@task(cache_version="1.0", cache=True, limits=Resources(mem="200Mi"))
+def tsne_and_plot(model_ser: FlyteFile[MODELSER_NLP]) -> plotdata:
+    model = Word2Vec.load(model_ser.path)
+    num_dimensions = 2
+    vectors = np.asarray(model.wv.vectors)
+    labels = np.asarray(model.wv.index_to_key)
+    tsne = TSNE(n_components=num_dimensions, random_state=0)
+    vectors = tsne.fit_transform(vectors)
+    x_vals = [v[0] for v in vectors]
+    y_vals = [v[1] for v in vectors]
+    plot_with_matplotlib(x_vals, y_vals, labels)
+    return x_vals, y_vals, labels
+
+
+# %%
+# Running the Workflow
+# ========
+#
+# Next, we define a workflow to call the aforementioned tasks.
+# This will return both models as workflow outputs
```

Comment:
```suggestion
# Lastly we define a workflow to call the aforementioned tasks.
# This will return the Word2Vec and LDA models as workflow outputs.
```